public/diagnostic.js

```diff
@@ -23,3 +23,26 @@
 //  -  Order 1 : 2 hats ($5 each)
 //  -  Order 2 : 1 desk lamp ($20 each)
 //  -  Order 3 : 3 hand towels ($8 each)
+
+
+
+const Cart =  Ember.Object.extend({
+  orders: [],
+  addToCart(item) {
+    this.orders.push(item);
+    return ("Added " + item.description + "to the cart.");
+  },
+
+  totalPrice: Ember.computed('orders.@each.price', function() {
+    let sum = orders.map((order) => order.price * order.quantity).reduce((a,b) => a+b);
+    return sum;
+  })
+});
```

Comment:
this solution for `totalPrice` is close.  instead of `order.price * order.quantity` you would use `order.get('orderPrice')` referring to the method that the Order class needs.